plugins/utils/yanky.nix

```diff
@@ -174,80 +302,50 @@ in
     };
   };
 
-  config =
-    let
-      thereAreSomeTelescopeMappings =
-        (cfg.picker.telescope.mappings != null) && (cfg.picker.telescope.mappings != { });
-    in
-    mkIf cfg.enable {
-      assertions = [
-        {
-          assertion = thereAreSomeTelescopeMappings -> cfg.picker.telescope.enable;
-          message = ''
-            Nixvim: You have defined some mappings in `plugins.yanky.picker.telescope.mappings` but
-            you have not set `plugins.yanky.picker.telescope.enable`.
-          '';
-        }
-      ];
+  settingsExample = {
+    highlight = {
+      on_put = true;
+      on_yank = true;
+      timer = 500;
+    };
+    preserve_cursor_position = {
+      enabled = true;
+    };
+    picker = {
+      telescope = {
+        mappings = {
+          default = "require('yanky.telescope.mapping').put('p')";
+          i = {
+            "<c-g>" = "require('yanky.telescope.mapping').put('p')";
+            "<c-k>" = "require('yanky.telescope.mapping').put('P')";
+            "<c-x>" = "require('yanky.telescope.mapping').delete()";
+            "<c-r>" = "require('yanky.telescope.mapping').set_register(require('yanky.utils').get_default_register())";
+          };
+          n = {
+            p = "require('yanky.telescope.mapping').put('p')";
+            P = "require('yanky.telescope.mapping').put('P')";
+            gp = "require('yanky.telescope.mapping').put('gp')";
+            gP = "require('yanky.telescope.mapping').put('gP')";
+            d = "require('yanky.telescope.mapping').delete()";
+            r = "require('yanky.telescope.mapping').set_register(require('yanky.utils').get_default_register())";
```

Comment:
I agree this would be nicer with the locally scoped variables you suggested, but I don't think we _need_ to decide on this PR (they can be added any time).

I also have no strong opinion, but if we do add them I think we should avoid removing them, as that could break user's configs.